src/msg.c

```diff
@@ -715,8 +722,10 @@ static ssize_t msg_process(struct session *session,
 	case KNOT_MSG_REG_REQ:
 		/* Payload length is set by the caller */
 		result = msg_register(session, &kreq->reg, ilen, &krsp->cred);
-		if (result != 0)
+		if (result != 0) {
+			rtype = KNOT_MSG_REG_RSP;
```

Comment:
Here you just assigning the `rtype` if `result != 0`. Or you change in each case  msg_schm_frag, msg_schm_end, msg_auth or put it out of scope of the `if` statement

Reply:
Done